src/avdecc/mcDomainManager.cpp

```diff
@@ -0,0 +1,443 @@
+			{
+				auto secondaryMasterIdKV = getMediaClockMaster(entityId); // get second connection if there is one.
+				if (secondaryMasterIdKV.first) // check if the id is valid
+				{
+					associatedDomains.push_back(secondaryMasterIdKV.first);
+					if (secondaryMasterIdKV.second == Error::NoError)
+					{
+						auto domainIndex = result.findDomainIndexByMasterEntityId(secondaryMasterIdKV.first);
+						if (!domainIndex)
+						{
+							// domain not created yet
+							MCDomain mediaClockDomain;
+							mediaClockDomain.setMediaClockDomainMaster(secondaryMasterIdKV.first);
+							domainIndex = result.mediaClockDomains().size();
+							mediaClockDomain.setDomainIndex(domainIndex.value());
+							result.mediaClockDomains().emplace(domainIndex.value(), mediaClockDomain);
+						}
+						associatedDomains.push_back(domainIndex.value());
+					}
+				}
+			}
+			result.entityMediaClockMasterMappings().emplace(entityId, associatedDomains);
+		}
+
+		return result;
+	}
+
+private:
+	/**
+* Finds the index of the clock stream.
+* @return If the clock stream index can not be determined,  std::numeric_limits<la::avdecc::entity::model::StreamIndex>().max() is returned.
+*/
+	std::optional<la::avdecc::entity::model::StreamIndex> findClockStreamIndex(la::avdecc::controller::model::ConfigurationNode const& configNode)
+	{
+		for (auto const& streamInputKV : configNode.streamInputs)
+		{
+			auto const& streamInput = streamInputKV.second;
+			for (auto const& streamFormat : streamInput.staticModel->formats)
+			{
+				auto const streamFormatInfo = la::avdecc::entity::model::StreamFormatInfo::create(streamFormat);
+				auto streamType = streamFormatInfo->getType();
+				if (la::avdecc::entity::model::StreamFormatInfo::Type::ClockReference == streamType)
+				{
+					return streamInputKV.first;
+				}
+			}
+		}
+		return std::nullopt;
+	}
+
+	/**
+* Compares the current mc mapping state with the previous one and 
+* emits the mediaClockConnectionsUpdate to inform the views about changes.
+*/
+	void notifyChanges()
+	{
+		// potenitally changes every other entity...
+		std::vector<la::avdecc::UniqueIdentifier> changes;
+		auto currentMCDomainMapping = createMediaClockDomainModel();
+		for (auto const& entityDomainKV : currentMCDomainMapping.entityMediaClockMasterMappings())
+		{
+			la::avdecc::UniqueIdentifier const domainMasterId = entityDomainKV.first;
+			std::vector<DomainIndex> domainIds = entityDomainKV.second;
+			if (_currentMCDomainMapping.entityMediaClockMasterMappings().count(domainMasterId))
+			{
+				auto oldDomainIndexes = _currentMCDomainMapping.entityMediaClockMasterMappings().at(domainMasterId);
+				if (oldDomainIndexes.size() > 0 && domainIds.size() > 0)
+				{
+					auto oldMaster = _currentMCDomainMapping.mediaClockDomains().at(oldDomainIndexes.at(0)); // the first index is the mc master
+					auto newMaster = currentMCDomainMapping.mediaClockDomains().at(domainIds.at(0)); // the first index is the mc master
```

Comment:
Can you please explain the comparison algorithm used here, because I'm a bit lost. Why only checking the first index?
Why not just compare the vectors?

Reply:
The first entry is the only relevant here. The second entry occurs in the special case of the entity being it's own mc master (clock source set to internal) and also having a clock stream connection. This will be relevant in a future feature.

Reply:
For the sake of readability, it would be best to create a new comparison function, taking both vectors and returning a bool. It will then be easier to:
* comment the comparison algo
* update it when needed